modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Field.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2021-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.iglu.schemaddl.parquet
+
+import com.snowplowanalytics.iglu.schemaddl.StringUtils
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.Schema
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.properties.{ArrayProperty, CommonProperties}
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.mutate.Mutate
+
+case class Field(name: String, 
+                 fieldType: Type, 
+                 nullability: Type.Nullability)
+
+object Field {
+
+  def build(name: String, topSchema: Schema, enforceValuePresence: Boolean): Field = {
+    val constructedType = buildType(Mutate.forStorage(topSchema))
```

Comment:
How about having separate pair of `build...` methods which do not mutate? I think it could be useful on a lib consumer side when mutation behaves unexpectedly (hopefully not) and then quickly switch to build without mutation

Reply:
Great question!  I made this decision very deliberately.  There's two reasons why I enforced mutating before converting to parquet.

First, we want all of our parquet files to be compatible with each other.  Currently we only use these methods in the rdb-transformer, but one day we might have other apps that need to write parquet files from Iglu+Json.  And I want to remove the possibility that a developer accidentally calls the wrong method, or try converting to parquet without mutating first.

Secondly, we've talked recently at Snowplow about being more opinionated when we implement something.  Instead of making something configurable (mutate or don't mutate) and then we have to forever support both versions.  Surprisingly, most of the time, users trust our judgement and like to be told the best way to do things.  So there's often there's no reason to provide two different ways of doing something.

That said.... if in future we have a solid reason to split out a non-mutating field build, then I'd be happy to provide the method when we need it.